### PR TITLE
8270282: Semantically rename reference processing subphases

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple rename of phase{2,3,4} to something with semantic meaning.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8270282](https://bugs.openjdk.java.net/browse/JDK-8270282): Semantically rename reference processing subphases

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to 49a4e277a3b16f5340ab1142205686e835e544b3
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to 49a4e277a3b16f5340ab1142205686e835e544b3

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4752/head:pull/4752` \
`$ git checkout pull/4752`

Update a local copy of the PR: \
`$ git checkout pull/4752` \
`$ git pull https://git.openjdk.java.net/jdk pull/4752/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4752`

View PR using the GUI difftool: \
`$ git pr show -t 4752`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4752.diff">https://git.openjdk.java.net/jdk/pull/4752.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/referenceProcessorPhaseTimes.hpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ class ReferenceProcessorPhaseTimes : public CHeapObj<mtGC> {
   ReferenceProcessorPhaseTimes(GCTimer* gc_timer, uint max_gc_threads);
   ~ReferenceProcessorPhaseTimes();
 
-  WorkerDataArray<double>* phase2_worker_time_sec() const { return _phase2_worker_time_sec; }
+  WorkerDataArray<double>* SoftWeakFinalRefsPhase_worker_time_sec() const { return _SoftWeakFinalRefsPhase_worker_time_sec; }
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/referenceProcessorPhaseTimes.hpp:83]: This is a weird function name.  I understand why, but I'm not convinced.  We have process_soft_weak_final_refs and the like, for example.

**Review by kimbarrett — COMMENTED** [review]
Mostly looks good, except for a naming choice.

File: src/hotspot/share/gc/shared/referenceProcessorPhaseTimes.hpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/shared/referenceProcessorPhaseTimes.hpp:83]: How about `soft_weak_final_refs_phase_worker_time_sec()`?

Comment (kimbarrett) [reply src/hotspot/share/gc/shared/referenceProcessorPhaseTimes.hpp:83]: Yes, that's normal style.  It's quite a mouthful, but it's not like it gets written very often.  With that change, I think this PR looks good.

**Review by kimbarrett — APPROVED** [review]
Looks good after discussed function name change.

File: src/hotspot/share/gc/shared/referenceProcessorPhaseTimes.hpp (hunk 1 above)
Comment (tschatzl) [reply src/hotspot/share/gc/shared/referenceProcessorPhaseTimes.hpp:83]: Agree with that.

**Review by tschatzl — APPROVED** [review]
Lgtm apart from that naming issue.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**